rust/src/tests/rmdir.rs

```diff
@@ -29,3 +29,44 @@ fn changed_time_parent_success(ctx: &mut TestContext) {
         });
     });
 }
+
+/// Dummy mountpoint to check that rmdir returns EBUSY when using it on a mountpoint.
+struct DummyMnt {
+    path: PathBuf,
+}
+
+impl DummyMnt {
+    fn new(ctx: &mut TestContext) -> anyhow::Result<Self> {
+        // We don't really care about a specific type of file system here, the directory just have to be a mount point
+        let from = ctx.create(crate::runner::context::FileType::Dir)?;
+        let path = ctx.create(crate::runner::context::FileType::Dir)?;
+        let mut mount = Command::new("mount");
+
+        if cfg!(target_os = "linux") {
+            mount.arg("--bind");
+        } else {
+            mount.args(["-t", "nullfs"]);
+        }
+
+        let result = mount.arg(&from).arg(&path).output()?;
+        debug_assert!(result.status.success());
```

Comment:
Why would you disable this assertion in release builds?
```suggestion
        assert!(result.status.success());
```

Reply:
I thought that if we get the flags right on the debug build, we wouldn't need to check on release. But you're right, it's better to keep them even on release.